src/interface/src/app/services/jwt-interceptor.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { catchError, Observable, switchMap } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class JwtInterceptor implements HttpInterceptor {
+  constructor(private auth: AuthService) {}
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(
+      catchError((error) => {
+        if (
+          error instanceof HttpErrorResponse &&
+          !request.url.includes('login') &&
```

Comment:
this might be `dj-auth-rest` instead of just login if we wanna exclude al 401 from dj-auth